### PR TITLE
MEN-4077: configuration file reading and validation

### PR DESCRIPTION
**Pull request by tranchitella**
Changelog: title

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+	// List of available servers, to which client can fall over
+	Servers []https.MenderServer
+	// The command to run as shell
+	ShellCommand string
+	// Name of the user who owns the shell process
+	User string
+}
+
+// MenderConfig holds the configuration settings for the Mender shell client
+type MenderConfig struct {
```
Comment (merlin-northern) [inline config/config.go:54]: I wonder: should we rename to MenderShellConfig?

File: client/https/config.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+// Security structure holds the configuration for the client
+// Added for MEN-3924 in order to provide a way to specify PKI params
+// outside HttpsClient.
+// NOTE: Careful when changing this, the struct is exposed directly in the
+// 'mender-shell.conf' file.
+type Security struct {
+	AuthPrivateKey string
+	SSLEngine      string
+}
```
Comment (merlin-northern) [inline client/https/config.go:54]: technically we do not need this section, unless we plan to sign some requests. no harm in keeping it.

**Review by merlin-northern — COMMENTED** [review]
looks good. I would seek all `/Mender[^a-zA-Z0-9]/` occurrences and change to `MenderShell`. two small comments.

File: client/https/config.go (hunk 2 above)
Comment (tranchitella) [reply client/https/config.go]: let's remove it, I did not know it is only used when signing requests

File: config/config.go (hunk 1 above)
Comment (tranchitella) [reply config/config.go]: renamed

**Review by merlin-northern — APPROVED** [review]
looks good to me!

**Pull request merged**